public/app/features/logs/components/InfiniteScroll.tsx

```diff
@@ -178,13 +178,15 @@ function getVisibleRange(rows: LogRowModel[]) {
 }
 
 function getPrevRange(visibleRange: AbsoluteTimeRange, currentRange: TimeRange) {
-  return { from: currentRange.from.valueOf(), to: visibleRange.from };
+  // Until the oldest log line, but without including it
+  return { from: currentRange.from.valueOf(), to: visibleRange.from - 1 };
```

Comment:
Hmm, difficult. 1 millisecond seems to be quite long, especially for Loki that is based on nanoseconds. I know it might be performance heave, but couldn't we do a similar approach to 

https://github.com/grafana/grafana/blob/0f8d4db934a0428afc1e6142be3b32a3f9b45b20/public/app/features/logs/components/log-context/LogRowContextModal.tsx

Reply:
Yeah, good point. We definitely need to be able to use ns on requests 😬